execution_engine/src/engine_state/mod.rs

```diff
@@ -686,35 +684,31 @@ where
     /// For a given root [`Key`] it does a path lookup through the named keys.
     ///
     /// Returns the value stored under a [`URef`] wrapped in a [`QueryResult`].
-    pub fn run_query(&self, query_request: QueryRequest) -> Result<QueryResult, Error> {
-        let tracking_copy = match self.tracking_copy(query_request.state_hash())? {
-            Some(tracking_copy) => Rc::new(RefCell::new(tracking_copy)),
-            None => return Ok(QueryResult::RootNotFound),
+    pub fn run_query(&self, query_request: QueryRequest) -> QueryResult {
+        let state_hash = query_request.state_hash();
+        let query_key = query_request.key();
+        let query_path = query_request.path();
+        let query_result = match self.tracking_copy(state_hash) {
+            Ok(Some(tc)) => match tc.query(query_key, query_path) {
+                Ok(ret) => ret.into(),
+                Err(err) => QueryResult::Failure(err),
+            },
+            Ok(None) => QueryResult::RootNotFound,
+            Err(err) => QueryResult::Failure(err),
         };
 
-        let tracking_copy = tracking_copy.borrow();
-
-        match tracking_copy.query(query_request.key(), query_request.path()) {
-            Ok(TrackingCopyQueryResult::ValueNotFound(result_string)) => {
-                let key = query_request.key();
-
-                if !key.is_system_key() {
-                    return Ok(TrackingCopyQueryResult::ValueNotFound(result_string).into());
+        if let QueryResult::ValueNotFound(_) = query_result {
```

Comment:
We need to get rid of this block, cause of the pruning changes during the boundary upgrade